drivers/bbram/bbram_stm32.c

```diff
@@ -14,7 +14,14 @@
 LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
 
 #define STM32_BKP_REG_BYTES		 4
+#ifdef CONFIG_SOC_SERIES_STM32WLX
+/* STM32WL series has backup registers defined in the TAMP section, not in the
+ * RTC section.
+ */
+#define STM32_BKP_REG_OFFSET		 (TAMP_BASE + offsetof(TAMP_TypeDef, BKP0R) - RTC_BASE)
+#else
 #define STM32_BKP_REG_OFFSET		 0x50
```

Comment:
Isn't there a way to defined child's register address in a proper way in both cases which would avoid this ?

Reply:
This is the implementation on stm32wbxx:
```
__STATIC_INLINE void LL_RTC_BAK_SetRegister(RTC_TypeDef *RTCx, uint32_t BackupRegister, uint32_t Data)
{
  __IO uint32_t *tmp;

  tmp = &(RTCx->BKP0R) + BackupRegister;

  *tmp = Data;
}
```

This is the implementation on stm32wlxx:
```
__STATIC_INLINE void LL_RTC_BKP_SetRegister(RTC_TypeDef *RTCx, uint32_t BackupRegister, uint32_t Data)
{
  uint32_t tmp;

  UNUSED(RTCx);

  tmp = (uint32_t)(&(TAMP->BKP0R));
  tmp += (BackupRegister * 4U);

  /* Write the specified register */
  *(__IO uint32_t *)tmp = (uint32_t)Data;
}
```

They don't use the same name. And on stm32wlxx, the context pointer is unused.

Now that you've pointed it out, I see that the problem is more widespread than just stm32wl. I misunderstood the ST range.

HAL that uses `LL_RTC_BKP_SetRegister`, have backup registers in `TAMP`:
- f1
- g0
- g4
- h5
- h7
- l4
- l5
- mp1
- u5
- wba
- wl

HAL that uses `LL_RTC_BAK_SetRegister`, have backup registers in `RTC`:
- f0
- f2
- f3
- f4
- f7
- h7
- l0
- l1
- l4
- wb

Both functions appear in some HAL:
- in h7 the test uses `#if !defined(TAMP)`.
- In l4, the test uses `#if defined(RTC_BACKUP_SUPPORT)`.

I'm having a hard time finding a suitable variable to identify the location of the registers. Also, could there be a MCU with backup registers both in RTC and in TAMP ?

Reply:
This may be many dts to rework and it would be more error prone to calculate and type in all these addresses than to rely on preprocessor macro. I'm note sure delegating a configuration responsibility to the dts is the best move, if there are indeed only two cases.

Reply:
In general, we should aim for the DT description solution when possible.
Though I don't have ready to use dt description for this and this remains a single switch using series abstraction.

So it is ok for now.